tools/ncf_clone_test.sh

```diff
@@ -11,44 +10,114 @@
 TEST_DEPENDENCIES="curl lsb-release"
-BRANCH="$1"
 
-# Disable any internationalization
-LC_ALL=C
-unset LANG
+# Default values
+BRANCH="master"
+REPO="https://github.com/Normation/ncf.git"
+
+# CFEngine source to use
+# can be:
+# "rudder" for a Rudder agent
+# "cfengine" for a CFEngine agent
+SOURCE="$1"
 
-## Runtime
+# Version to use
+# Rudder versions use the same notation as rtf
+# CFEngine versions must by of the x.y.z form
+VERSION="$2"
 
-# 1 - Install the dependencies
+# ncf branch to test
+if [ "z$3" != "z" ]; then
+	BRANCH="$3"
+fi
+
+# Repository to clone
+if [ "z$4" != "z" ]; then
+	REPO="$4"
+fi
+
+# Documentation
+usage() {
+  echo "Usage $0 [rudder|cfengine] version [branch] [repository]"
+	echo ""
+  echo "  Setup an agent and starts the ncf tests. Should be launched only on disposable test machines"
+	echo "  as some of the tests may modify the system."
+	echo ""
+	echo "  rudder|cfengine: binary package source"
+	echo ""
+	echo "  version: version to use, for example"
+	echo "    for cfengine source: 3.6.5, 3.7.3"
+	echo "    for rudder source: 3.1, 3.0.10, ci/3.2.14, 3.1-nightly"
+	echo ""
+	echo "  branch: ncf branch to use (default: master)"
+	echo ""
+	echo "  repository: repository to clone (default: https://github.com/Normation/ncf.git)"
+  exit 1
+}
+
+install_rudder_agent() {
+	cd ${TEST_WORKDIR}
+	
+	curl -sO "https://www.rudder-project.org/tools/rudder-setup"
+	sh rudder-setup setup-agent ${VERSION}
+	
+	# Put CFEngine binaries in the default PATH
+	export PATH="${PATH}:/var/rudder/cfengine-community/bin/"
+}
+
+install_cfengine_agent() {
+	cd ${TEST_WORKDIR}
+	
+	if type dpkg >/dev/null 2>/dev/null; then
+	  curl -s0 -o cfengine-community.deb "https://cfengine-package-repos.s3.amazonaws.com/community_binaries/cfengine-community_${VERSION}-1_amd64.deb"
+		dpkg -i cfengine-community.deb
+	else
+		if type rpm >/dev/null 2>/dev/null; then
+	  	curl -s0 -o cfengine-community.rpm "https://cfengine-package-repos.s3.amazonaws.com/community_binaries/cfengine-community_${VERSION}-1.x86_64.rpm"
+			rpm -i cfengine-community.rpm
+		else
+			echo "Unsupported platform."
+			exit 1
+		fi
+  fi
+}
+
+if [ "$#" -ne 2 ] && [ "$#" -ne 3 ] && [ "$#" -ne 4 ]; then
+	usage
+fi
+
+# Install the dependencies
 apt-get update
 apt-get -y install ${TEST_DEPENDENCIES}
 
-# 2 - Install CFEngine from the Normation repository if necessary
-# Note: The second apt-get update is mandatory, need curl before adding the new apt source
-if ! dpkg-query -W --showformat='${Status}\n' cfengine-community | grep -q "install ok installed"
-then
-	curl http://www.normation.com/cfengine-repo/gpg.key | apt-key add -
-	echo "deb http://www.normation.com/cfengine-repo/apt $(lsb_release -cs) main" > /etc/apt/sources.list.d/cfengine-community.list
-	apt-get update
-	apt-get -y install cfengine-community
+# Install the agent
+if [ "${SOURCE}" = "rudder" ]; then
+	install_rudder_agent
+	echo "Using CFEngine binaries from Rudder ${VERSION} packages"
+else
```

Comment:
Could you add a sanity check here to make sure source=cfengine? This is important because $1 used to be the branch name and calling this script like "script v0.x" will work but not so what is expected.